kratos/python_scripts/save_restart_process.py

```diff
@@ -0,0 +1,75 @@
+from __future__ import print_function, absolute_import, division #makes KratosMultiphysics backward compatible with python 2.6 and 2.7
+# Importing the Kratos Library
+import KratosMultiphysics
+
+
+def Factory(settings, Model):
+    if(type(settings) != KratosMultiphysics.Parameters):
+        raise Exception("Expected input shall be a Parameters object, encapsulating a json string")
+    return SaveRestartProcess(Model, settings["Parameters"])
+
+
+class SaveRestartProcess(KratosMultiphysics.Process):
+
+    def __init__(self, model, params):
+        """This process compares saves restart files
+        It works both in OpenMP and MPI
+        see the "default_settings" for available options
+        """
+        ## Settings string in json format
+        default_settings = KratosMultiphysics.Parameters("""{
+            "model_part_name"              : "",
+            "echo_level"                   : 0,
+            "serializer_trace"             : "no_trace",
+            "restart_save_frequency"       : 0.0,
+            "restart_control_type"         : "time",
+            "save_restart_files_in_folder" : true
+        }""")
+
+        ## Overwrite the default settings with user-provided parameters
+        params.ValidateAndAssignDefaults(default_settings)
+        self.params = params
+        self.model = model
+
+        if self.params["model_part_name"].GetString() == "":
+            raise Exception('No "model_part_name" was specified!')
+
+    def ExecuteInitialize(self):
+        model_part = self.model[self.params["model_part_name"].GetString()]
+
+        is_mpi_execution = (model_part.GetCommunicator().TotalProcesses() > 1)
+
+        if is_mpi_execution:
+            import KratosMultiphysics.TrilinosApplication
+            from trilinos_restart_utility import TrilinosRestartUtility as Restart
+        else:
+            from restart_utility import RestartUtility as Restart
+
+        self.params.AddValue("input_filename", self.params["model_part_name"])
+        self.params.RemoveValue("model_part_name")
+
+        self.restart_utility = Restart(model_part, self.params)
+
+    def ExecuteBeforeSolutionLoop(self):
+        pass
+
+    def ExecuteInitializeSolutionStep(self):
+        pass
+
+    def ExecuteFinalizeSolutionStep(self):
+        pass
+
+    def ExecuteBeforeOutputStep(self):
+        pass
+
+    def IsOutputStep(self):
+        return self.restart_utility.IsRestartOutputStep()
+
+    def PrintOutput(self):
+        self.restart_utility.SaveRestart()
+
+    def ExecuteAfterOutputStep(self):
+        pass
+
+    def ExecuteFinalize(self):
+        pass
```

Comment:
Minor suggestion (maybe for a future PR): When I have had to use restart, it was often because I was running on a cluster with some limit on run-time for the cases. In such situations, I would just estimate how many time steps I could run in a single batch, launched a job for that and ensured that the model part was serialized at the end of the job. I think it would make sense that the process allows this, which could be achieved by having an extra save on `ExecuteFinalize` (maybe only if explicily requested via json?).

Reply:
Sounds fine to me, we can speak tmr abt this